### ISSUE TITLE
Navigate to Thread Messages When Clicking on Thread Message PNs

### ISSUE THREAD

**Issue by MarinTolic** [GetStream/stream-chat-android#4608]
### Context & proposal

Currently, our SDK does not navigate to messages inside threads when such messages have been received as push notifications and opened via aforementioned notifications.

We should update our SDK so that it automatically recognizes thread messages and opens them accordingly.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Implement navigating to thread messages in the V5 XML SDK
- [ ] Implement navigating to thread messages in the V5 Compose SDK
- [ ] Implement navigating to thread messages in the V6 XML SDK
- [ ] Implement navigating to thread messages in the V6 Compose SDK

**Comment by MarinTolic** [comment]
Soft blocked by #4616 .

During a pair programming session #4616 was found to be useful in bettering the reliability of the goals stated in the current ticket. 

#4616 is being scheduled for work immediately and can be considered a part of the current effort, it has simply been moved to a separate ticket so we don't bloat a single PR with too many changes which are connected but not necessarily tightly coupled and can be separated into readable PRs.
